src/main/java/sirius/pasta/noodle/compiler/ir/ConstructorCall.java

```diff
@@ -100,12 +100,18 @@ private void checkDeprecation(CompilationContext context) {
     }
 
     private void checkSandbox(CompilationContext context) {
-        if (context.isSandboxEnabled() && !sandbox.canInvoke(constructor)) {
-            context.error(position,
-                          "Cannot invoke the constructor on %s as this is restricted by the security sandbox!",
-                          constructor.getDeclaringClass().getName());
-            this.constructor = null;
-        }
+        context.ifEnforceSandbox(() -> !sandbox.canInvoke(constructor), mode -> {
+            if (mode == SandboxMode.ENABLED) {
```

Comment:
If not overkill, can we generalize this? Feels like the same check is used in several places to either call context.error x context.warning.

Reply:
Sadly the three usages all differ each a little bit (having side-effects on error etc.)